src/librbd/internal.cc

```diff
@@ -113,6 +121,18 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
         }
 
         r = remote_request();
+        if (r == -EOPNOTSUPP && request_lock) {
+          AioCompletion *comp = aio_create_completion();
+          ictx->image_watcher->request_lock(boost::bind(&complete_request, _1),
```

Comment:
@dillaman Is it worth adding a new type on hammer branch? Also, additionally I would need to call comp->init_time(...) to set the type. If you think it is ok, I can do this way though.